Add use_jit option to URDFparser interface

cmower opened the pull request:
Hi, first, nice package! However, it didn't work out-of-the-box for me - even `pip install urdf2casadi` doesn't work. I believe the issue is that you use `jit` options by default when instantiating the `URDFparser` class for function objects. This requires casadi to be built with the `--WITH_CLANG` option on.

Can I suggest you merge this commit, that just defaults to no options instead.

mahaarbo commented:
Hi!

If you instantiate the URDFParser object with `func_opts={}`, the jit options will be removed which is probably an easy fix in your case. Which OS are you running? On ubuntu, I've used `pip install casadi` and the jit functionality without needing any additional build options.

cmower commented:
I am running Ubuntu 20.04 and have my own custom build of casadi - you are correct, `pip install casadi` comes with jit functionality, I didn't build with this option since I have no need. So as you suggest, `func_opts={}` resolves the issue.

However, I still suggest you merge the changes since it was a headache to find out that this was the issue. Also, I don't agree with changing default interface options when writing a wrapper to a method/library. The changes won't break anything for other users whether they have a custom build or installed via `pip`. 

But, in the end, your decision :smile:

mahaarbo commented:
> However, I still suggest you merge the changes since it was a headache to find out that this was the issue. Also, I don't agree with changing default interface options when writing a wrapper to a method/library. The changes won't break anything for other users whether they have a custom build or installed via `pip`.

Good point regarding changing the default interface. However the library is already in use by others and I fear that the experience "out of the box" for users may change if it suddenly changes to non-jit compiled functions.

Perhaps a compromise would be to add an optional flag `use_jit` on the init function that defaults to true and is documented in the init as a potential source of error if casadi is not built with clang?

cmower commented:
> Perhaps a compromise would be to add an optional flag `use_jit` on the init function that defaults to true and is documented in the init as a potential source of error if casadi is not built with clang?

Sounds good to me, I've updated the PR, resolved conflicts, and implemented your suggestion - should be ok to merge?

mahaarbo commented:
Looks good! Thanks for the contribution!
